Diagnostic/Providers/Builtin.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+# MIT License
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Utils.ProviderUtil as ProvUtil
+from collections import defaultdict
+
+# These are the built-in metrics this code provides, grouped by class.
+_builtIns = {
+    'processor' : [ 'percentidletime', 'percentprocessortime', 'percentiowaittime', 'percentinterrupttime',
+                    'percentusertime', 'percentnicetime', 'percentprivilegedtime' ],
+    'memory':     [ 'availablememory', 'percentavailablememory', 'usedmemory', 'percentusedmemory', 'pagespersec',
+                    'pagesreadpersec', 'pageswrittenpersec', 'availableswap', 'percentavailableswap', 'usedswap',
+                    'percentusedswap' ],
+    'network':    [ 'bytestransmitted', 'bytesreceived', 'bytestotal', 'packetstransmitted', 'packetsreceived',
+                    'totalrxerrors', 'totaltxerrors', 'totalcollisions' ],
+    'filesystem': [ 'freemegabytes', 'usedmegabytes', 'percentfreespace', 'percentusedspace', 'percentfreeinodes',
+                    'percentusedinodes', 'bytesreadpersecond', 'byteswrittenpersecond', 'bytespersecond',
+                    'readspersecond', 'writespersecond', 'transferspersecond' ],
+    'disk':       [ 'readspersecond', 'writespersecond', 'transferspersecond', 'averagereadtime', 'averagewritetime',
+                    'averagetransfertime', 'averagediskqueuelength', 'readbytespersecond', 'writebytespersecond',
+                    'bytespersecond' ]
+    }
+
+_omiClassName = { 'processor': 'SCX_ProcessorStatisticalInformation',
+                  'memory': 'SCX_MemoryStatisticalInformation',
+                  'network': 'SCX_EthernetPortStatistics',
+                  'filesystem': 'SCX_FileSystemStatisticalInformation',
+                  'disk': 'SCX_DiskDriveStatisticalInformation'
+                }
+
+_instancedClasses = [ 'network', 'filesystem', 'disk', 'processor' ]
```

Comment:
Because memory isn't instanced? You can have multiple processors, multiple disks, multiple filesystems, multiple networks... only one memory instance. That's what "instanced" means in the OMI context.